core/parser.mly

```diff
@@ -171,6 +171,9 @@ let parseRegexFlags f =
               | 'g' -> RegexGlobal
               | _ -> assert false) (asList f 0 [])
 
+let fresh_typevar freedom = ("_", None, freedom)
+let fresh_effects = ([], Datatype.Open ("_anon", None, `Rigid))
```

Comment:
I wonder whether the name `_anon` can ever leak out? I suppose it can't, because the name `_anon` is compiled away before the semantic type is constructed, and the type printer refreshes the type variable names anyway. And the error reporter for syntax errors extracts excerpts from the actual source code, in which `_anon` isn't present.